flink-runtime-web/web-dashboard/src/app/pages/task-manager/log-list/task-manager-log-list.component.html

```diff
@@ -29,14 +29,16 @@
     <thead>
       <tr>
         <th>Log Name</th>
-        <th>Size (KB)</th>
+        <th [nzSortFn]="sortLastModifiedTimeFn">Last Modified Time</th>
+        <th [nzSortFn]="sortSizeFn">Size (KB)</th>
       </tr>
     </thead>
     <tbody>
-      <tr *ngFor="let log of listOfLog">
+      <tr *ngFor="let log of table.data">
```

Comment:
What type is inferred for `table.data`? If this is just inferred to be `any`, then this would be a regression in type safety.

Reply:
Thanks for review code firstly.  
I think table.data's type here is `TaskManagerLogItemInterface[]` and table.data's type in `job-manager-log-list.component.html` is `JobManagerLogItemInterface[]`. 
I read the doc of [NG_ZORRO#table](https://ng.ant.design/components/table/en#components-table-demo-basic) , and it's said that table.data's value is set by nz-table's property `[nzData]`, so I passed `listOfLog` to `nzData` which is defined in related .ts file .

Reply:
I don't mean what the type "should" be, but rather what the Angular Language Service will _actually_ infer. The last time I checked, Angular cannot use component generics to infer the type correctly. You can hover over the `log` in `let log` and see the type the language service will infer. If this is `any`, that's not good.

Reply:
I got it and I found that the type of `log` in  `let log in table.data` is `unknown` and  type of `table.data` is `unknown[]` by the way you said in IDE. I should change it to `let log in listOfLog` and then type of `log` will be clear and correct, am I right ? @Airblader

Reply:
If the change to `table.data` isn't needed, I would prefer leaving it as `listOfLog` for now, yes. If we need the change (like for pagination) then we should at least use the same type trick I mentioned [here](https://github.com/apache/flink/pull/17619#discussion_r741033701).

Reply:
I found that change to `listOfLog` will lead sortFn not work, and if I add `trackBy` function in ngFor loop, the type of `log` will be clear and correct. I'm not familiar with Angular, so I'm not sure if it's a type trick or not. However add trackBy function seems like a way that works.

Reply:
@Airblader how do you think of add trackBy function to help angular infer correct type ?

Reply:
`trackBy` is a useful addition regardless; if it happens to make the type inference work better as well, then that's fine with me.

Reply:
@Airblader I added trackBy function and this pr is ready to be reviewed again.
And please help me to rerun the CI , latest CI check was failed cause by flink-connector-rabbitmq 's integration-tests which is occasionally I think, thanks !